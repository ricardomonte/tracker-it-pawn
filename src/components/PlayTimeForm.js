import React, { useState } from 'react';
import PropTypes from 'prop-types';
import TrackerStyle from '../styles/PlayTimeForm.module.css';

const PlayTimeForm = ({ id, UpdatePlay }) => {
  const [hours, setHours] = useState('');

  const handleChange = (event) => {
    const updateHours = event.target.value;
    setHours(updateHours);
  };

  const handleValidation = () => {
    if (hours === '' || hours === '0') {
      return;
    }
    const updateDog = {
      id,
      playTime: parseInt(hours, 10),
    };
    UpdatePlay(updateDog);
  };

  return (
    <div className={TrackerStyle.containerTotal}>
      <form onSubmit={handleValidation} className={TrackerStyle.container}>
        <label htmlFor="hour" className={TrackerStyle.label}>
          How much time you spend playing weakly with your dog (aprox):
          <input className={TrackerStyle.inputs} id="hour" type="number" value={hours} name="hour" onChange={handleChange} placeholder="0" />
        </label>
        <button className={TrackerStyle.btn} type="submit">Submit</button>
      </form>
    </div>
  );
};

PlayTimeForm.propTypes = {
  id: PropTypes.string.isRequired,
  UpdatePlay: PropTypes.func.isRequired,
};

export default PlayTimeForm;
