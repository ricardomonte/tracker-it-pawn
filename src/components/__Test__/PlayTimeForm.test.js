import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import MockData from '../../../tools/MockData';
import PlayTimeForm from '../PlayTimeForm';

test('Content from Expense matches snapshot', () => {
  const content = render(<PlayTimeForm id={(MockData[0].id).toString()} UpdatePlay={jest.fn()} />);
  expect(content.container).toMatchSnapshot();
  expect(screen.getByLabelText('How much time you spend playing weakly with your dog (aprox):')).toBeInTheDocument();
});
