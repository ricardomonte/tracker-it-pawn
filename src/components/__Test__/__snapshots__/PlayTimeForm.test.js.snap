// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content from PlayTimeForm matches snapshot 1`] = `
<div>
  <div
    class="containerTotal"
  >
    <form
      class="container"
    >
      <label
        class="label"
        for="hour"
      >
        How many hours weekly do you spend playing with your dog (aprox):
        <input
          class="inputs"
          id="hour"
          name="hour"
          placeholder="0"
          type="number"
          value=""
        />
      </label>
      <button
        class="btn"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
